docker.go

```diff
@@ -42,72 +42,123 @@ func generatePort(portStr string) (nat.Port, error) {
 	return port, nil
 }
 
-// startContainer starts a docker container and returns the container ID
-// as well as a websocket connection to the attach endpoint.
-func (h *handler) startContainer(ctrInfo containerInfo) (string, *websocket.Conn, error) {
-	image := ctrInfo.dockerImage
-	// Use default docker image when user does not provide any
-	if image == "" {
-		image = defaultDockerImage
+type containerOptions func(cfg *container.Config)
+
+func withPort(port nat.Port) containerOptions {
+	return func(ctrCfg *container.Config) {
+		if port == "" {
+			return
+		}
+		ctrCfg.ExposedPorts = nat.PortSet{
+			port: struct{}{},
+		}
+	}
+}
+
+func withDockerImage(image string) containerOptions {
+	return func(cfg *container.Config) {
+		cfg.Image = image
+		if image == "" {
+			// Use default docker image when user does not provide any
+			cfg.Image = defaultDockerImage
+		}
 	}
+}
 
-	// pull container image if we don't already have it
-	if err := h.pullImage(image); err != nil {
-		return "", nil, fmt.Errorf("pulling %s failed: %v", image, err)
+type hostOptions func(cfg *container.HostConfig)
+
+func withExposedPort(port nat.Port) hostOptions {
+	return func(cfg *container.HostConfig) {
+		if port == "" {
+			return
+		}
+		cfg.PortBindings = map[nat.Port][]nat.PortBinding{
+			port: []nat.PortBinding{
+				{
+					HostIP:   "0.0.0.0",
+					HostPort: string(port),
+				},
+			},
+		}
+	}
+}
+
+func withSecurityOptions(seccompProfile string) hostOptions {
```

Comment:
Could the function be named `withSeccompOptions`?

Reply:
I think the name is valid because we are setting `SecurityOpt` of the `container.HostConfig` object. And since Seccomp profile is the only thing we are currently customizing hence that is the only parameter to the function.

https://github.com/kinvolk/contained.af/pull/6/files#diff-d3a679229fa5a654b0e1d154531ac1e1R93

Reply:
ok!